acvm/src/pwg/hash.rs

```diff
@@ -72,3 +73,42 @@ fn generic_hash_256<D: Digest>(
     let result = hasher.finalize().as_slice().try_into().unwrap();
     Ok(result)
 }
+
+pub fn keccak256(
+    initial_witness: &mut BTreeMap<Witness, FieldElement>,
+    gadget_call: &BlackBoxFuncCall,
+) -> Result<(), OpcodeResolutionError> {
+    generic_sha3::<Keccak256>(initial_witness, gadget_call)
+}
+
+fn generic_sha3<D: sha3::Digest>(
```

Comment:
This is the same `Digest` trait as used by `generic_hash_256` so we don't need a duplicate implementation ( you may need to adjust versions of dependencies)

Reply:
It is not the same because the two crates do not use the same version.

Reply:
Indeed, you'll need to adjust the dependency version for the blake2 crate to match.